server/src/main/java/org/opensearch/index/IndexService.java

```diff
@@ -867,7 +867,8 @@ public QueryShardContext newQueryShardContext(
         IndexSearcher searcher,
         LongSupplier nowInMillis,
         String clusterAlias,
-        boolean validate
+        boolean validate,
+        boolean keywordIndexOrDocValuesEnabled
```

Comment:
I had to pull this setting in here so that it can be accessed by `QueryShardContext`

Reply:
👍 

Thanks for overloading the method (so you don't break existing consumers)!